Header search path failure on 1.8.2 (works in 1.8.1) related to clang executable search order changes

I'm using autocxx to generate some bindings to a C++ library. autocxx [enables the `runtime` feature of this crate unconditionally](https://github.com/google/autocxx/issues/1276#issuecomment-2138955534), something which will hopefully be improved.

As of [this change](https://github.com/KyleMayes/clang-sys/commit/e46f0bf70a5ba9885788b0e5aa139c43bf8cebfd#diff-b063e256cdaf175454b7c43fa4321ba9ab479f0c41590773d91fb2fd23ac539bR50), released in 1.8.2, my bindings always fail to generate unless I set `CLANG_PATH=$(which clang)` to override the above search order, because the bundled libclang fails to find some macOS system headers.

```console
$ cargo update --precise 1.8.1 clang-sys
$ cargo check
...
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1m 02s

$ cargo update --precise 1.8.2 clang-sys
$ cargo check
...
  process didn't exit successfully: `/Users/.../build-script-build` (exit status: 1)
  --- stderr
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/inttypes.h:21:15: fatal error: 'inttypes.h' file not found
```

I'm not sure if this is something to be fixed here or not, as I don't fully understand why the bundled libclang doesn't work. But I suspect this problem affects others on macOS too.

I'm encountered same issue on MacOS

`
$ cargo update --precise 1.8.2 clang-sys
$ cargo build
...

  --- stderr
  wrapper.h:1:10: fatal error: 'sys/param.h' file not found
...
  Unable to generate bindings: ClangDiagnostic("wrapper.h:1:10: fatal error: 'sys/param.h' file not found\n")
`

I have the same issue. 
```
error: failed to run custom build command for `timelib v0.3.4`

Caused by:
  process didn't exit successfully: `/Users/user/Source/my-api-rust/target/llvm-cov-target/debug/build/timelib-9d2eb476496a3c91/build-script-build` (exit status: 101)
  --- stderr
  shim/../ext/timelib/timelib.h:37:10: fatal error: 'stdlib.h' file not found
  thread 'main' panicked at /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/timelib-0.3.4/build.rs:29:10:
  failed to run bindgen: ClangDiagnostic("shim/../ext/timelib/timelib.h:37:10: fatal error: 'stdlib.h' file not found\n")
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: command `/Users/user/.rustup/toolchains/stable-x86_64-apple-darwin/bin/cargo test --no-run --message-format json-render-diagnostics --target-dir /Users/user/Source/my-api-rust/target/llvm-cov-target` exited with code 101
error: process didn't exit successfully: `/Users/user/.rustup/toolchains/stable-x86_64-apple-darwin/bin/cargo nextest run --manifest-path /Users/user/Source/my-api-rust/Cargo.toml --target-dir /Users/user/Source/my-api-rust/target/llvm-cov-target` (exit status: 101)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,9 +5,6 @@
 
 ## [1.8.2] - 2024-05-29
 
-### Changed
-- Improved `Clang::find` to first check directories related to the runtime-loaded `libclang` instance (if any)
-
 ### Fixed
 - Fixed linking to `libclang` on Windows with MSYS2
 - Fixed `Clang::find` to support both the `-target` and `--target` arguments

--- a/src/link.rs
+++ b/src/link.rs
@@ -100,8 +100,8 @@ macro_rules! link {
         /// A dynamically loaded instance of the `libclang` library.
         #[derive(Debug)]
         pub struct SharedLibrary {
-            pub(crate) library: libloading::Library,
-            pub(crate) path: PathBuf,
+            library: libloading::Library,
+            path: PathBuf,
             pub functions: Functions,
         }
 

--- a/src/support.rs
+++ b/src/support.rs
@@ -44,13 +44,11 @@ impl Clang {
     /// Returns a `clang` executable if one can be found.
     ///
     /// If the `CLANG_PATH` environment variable is set, that is the instance of
-    /// `clang` used. Otherwise, these directories are searched in order:
-    ///
-    ///   1. The supplied path (if provided)
-    ///   2. Sibling directories for the runtime-loaded `libclang` instance (if any)
-    ///   3. The directory returned by `llvm-config --bindir`
-    ///   4. The directory returned by `xcodebuild -find clang` (on macOS)
-    ///   5. The directories in the system's `PATH` environment variable
+    /// `clang` used. Otherwise, a series of directories are searched. First, if
+    /// a path is supplied, that is the first directory searched. Then, the
+    /// directory returned by `llvm-config --bindir` is searched. On macOS
+    /// systems, `xcodebuild -find clang` will next be queried. Last, the
+    /// directories in the system's `PATH` are searched.
     ///
     /// ## Cross-compilation
     ///
@@ -83,16 +81,6 @@ impl Clang {
 
         if let Some(path) = path {
             paths.push(path.into());
-        }
-
-        #[cfg(feature = "runtime")]
-        if let Some(library) = crate::get_library() {
-            if let Some(directory) = library.path().parent() {
-                paths.push(directory.into());
-                if let Some(parent) = directory.parent() {
-                    paths.push(parent.join("bin"));
-                }
-            }
         }
 
         if let Ok(path) = run_llvm_config(&["--bindir"]) {

--- a/tests/lib.rs
+++ b/tests/lib.rs
@@ -50,14 +50,3 @@ fn test_support_target() {
     let clang = support::Clang::find(None, args).unwrap();
     println!("{:?}", clang);
 }
-
-#[cfg(feature = "runtime")]
-#[test]
-fn test_support_runtime() {
-    load().unwrap();
-    let library = get_library().unwrap();
-    let clang = support::Clang::find(None, &[]).unwrap();
-    println!("Library path: {}", library.path().display());
-    println!("Clang path:   {}", clang.path.display());
-    unload().unwrap();
-}

